fix vp301 IPagedDataSource.FetchNextPage

AliveMen opened the pull request:
changed enumeration methods from
IEnumerable<IExportable> FetchNextPage();

to
bool Fetch();
IEnumerable<IExportable> Items;

AliveMen committed:
changed enumeration methods from
IEnumerable<IExportable> FetchNextPage();

to
bool Fetch();
IEnumerable<IExportable> Items;

artem-dudarev commented on Modules/vc-module-export/VirtoCommerce.ExportModule.Data/Services/DataExporter.cs:
```diff
@@ -65,35 +65,34 @@ public void Export(Stream stream, ExportDataRequest request, Action<ExportProgre
-                        var objectBatch = pagedDataSource.FetchNextPage();
-
-                        if (objectBatch == null)
+                        while (pagedDataSource.Fetch())
```
Probably this line should be used instead of `while (exportedCount < totalCount)`

yecli replied:
Agree, Fetch() should replace outer cycle (it is enumerator `bool MoveNext()` analog).

AliveMen committed:
Merge branch 'export/dev' into export/fix_vp301_IPagedDataSource

AliveMen committed:
changed enumeration methods from
IEnumerable<IExportable> FetchNextPage();

to
bool Fetch();
IEnumerable<IExportable> Items;
